### PR TITLE
Upgrade to latest jwt response

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Use jwt `user_url` instead of `meta.*`
- Use hypermedia urls where possible instead of hard-coded api paths
- Remove uneeded state from root component; it's a stateless function component now 😍

# Result For User

No user change, but code is better :)

---

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -25,144 +25,137 @@ import InstructorsOnly from './components/InstructorsOnly'
...
+          </BrowserRouter>
+        )
+      }}
+    </Request>
+  )
 }
 
 ReactDOM.render(
-  <Routes />,
+  <App />,
```
Comment (trevordmiller) [inline src/index.js:159]: Routes wasn't very accurate since it has other logic too

File: src/index.js [hunk 2]
```diff
@@ -25,144 +25,137 @@ import InstructorsOnly from './components/InstructorsOnly'
...
-                    {({data}) => (
-                      <div>
-                        <Overview instructor={data} />
-                      </div>
-                    )}
-                  </Request>
-                )}
+  return (
+    <Request url={userUrl}>
+      {({data}) => {
```
Comment (trevordmiller) [inline src/index.js:38]: `data` is a variable shadow here, as routes also have <Request /> and data. It works but would be good if there was a simple way to make it more clear; wish we could rename while destructuring (`{data as user}`). Might make a `dataName` prop to pass to requests so returned render callback param has a more specific name than just `data` depending on instance.

File: src/index.js [hunk 3]
```diff
@@ -25,144 +25,137 @@ import InstructorsOnly from './components/InstructorsOnly'
...
+
+        if(data && !data.instructor_url) {
```
Comment (joelhooks) [inline src/index.js:40]: How should we handle non-instructors? Today @zacjones93 needed access, but he's not an instructor. The immediate solution was to just make him an instructor. On the rails side we have "editors" and "reviewers" and of course "admins"

More "food for thought" than anything to do right now...

File: src/index.js [hunk 4]
```diff
@@ -25,144 +25,137 @@ import InstructorsOnly from './components/InstructorsOnly'
...
-                render={({match}) => (
-                  <Request url={`/api/v1/instructors/${match.params.slug}`}>
-                    {({data}) => (
-                      <Overview instructor={data} />
+                  />
+
+                  <Route 
+                    path={`/lessons/:slug`}
+                    render={({match}) => (
+                      <Request url={`/api/v1/lessons/${match.params.slug}`}>
```
Comment (joelhooks) [inline src/index.js:114]: this is where `uri_templates` would be nice in the responses. This would be on the user object.

**Review by joelhooks — APPROVED** [review]

File: src/index.js (hunk 3 above)
Comment (trevordmiller) [reply src/index.js:40]: Good question - I'm not sure; would be good to have some sort of data between egghead.io general user and full-on instructor though.

**Pull request merged**